plugin-build/src/main/kotlin/io/sentry/android/gradle/sourcecontext/CollectSourcesTask.kt

```diff
@@ -0,0 +1,142 @@
+import io.sentry.android.gradle.tasks.DirectoryOutputTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.*
+import org.gradle.workers.WorkAction
+import org.gradle.workers.WorkParameters
+import org.gradle.workers.WorkerExecutor
+import java.io.File
+import java.nio.file.Paths
+import javax.inject.Inject
+import org.gradle.api.Project
+import org.gradle.api.file.Directory
+import org.gradle.api.provider.Provider
+
+@CacheableTask
+abstract class CollectSourcesTask @Inject constructor(
+    private val workerExecutor: WorkerExecutor,
+    private val layout: ProjectLayout
+) : DirectoryOutputTask() {
+
+    init {
+        group = SENTRY_GROUP
+        description = "Collects sources into a single directory so they can be bundled together."
+    }
+
+    /** The Groovy source of the current project. */
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFiles
+    abstract val sourceFiles: ConfigurableFileCollection
+
+    @TaskAction fun action() {
+        workerExecutor.noIsolation().submit(CollectSourcesWorkAction::class.java) {
+            it.projectDir.set(layout.projectDirectory)
+            it.sourceFiles.from(this@CollectSourcesTask.sourceFiles)
+            it.output.set(this@CollectSourcesTask.output)
+        }
+    }
+
+    interface CollectSourcesParameters : WorkParameters {
+        val projectDir: DirectoryProperty
+        val sourceFiles: ConfigurableFileCollection
+        val output: DirectoryProperty
+    }
+
+    abstract class CollectSourcesWorkAction : WorkAction<CollectSourcesParameters> {
+
+        override fun execute() {
+            val outDir = parameters.output.getAndDelete()
+
+            println(parameters.sourceFiles.asFileTree.map { it.absolutePath })
+
+            val collectedSources = SourceCollector(
+                projectDir = parameters.projectDir.get().asFile,
+                sourceFiles = parameters.sourceFiles
+            ).collect()
+
+            collectedSources.forEach { collectedSource ->
+                println(collectedSource)
+                val targetFile = outDir.resolve(File(collectedSource.relativePath))
+                collectedSource.file.copyTo(targetFile, true)
+            }
+        }
+    }
+
+    companion object {
+        fun register(
+            project: Project,
+            sourceFiles: List<File>,
+            output: Provider<Directory>,
+            taskSuffix: String = ""
+        ): TaskProvider<CollectSourcesTask> {
+//        val oldAGPExtension = project.extensions.getByType(AppExtension::class.java)
+            return project.tasks.register("sentryCollectSources${taskSuffix}", CollectSourcesTask::class.java) { task ->
+//            task.sourceFiles.setFrom(oldAGPExtension.sourceSets.flatMap { it.java.getSourceFiles() })
```

Comment:
using `this.sourceSets.flatMap { it.javaDirectories.flatMap { project.fileTree(it) } }` instead as this sometimes returned empty